pkg/flags/flags.go

```diff
@@ -0,0 +1,139 @@
+	flagEtcdMemberUnknownThreshold  = "etcd-member-unknown-threshold"
+
+	// Compaction lease controller flags
+	flagEnableBackupCompaction = "enable-backup-compaction"
+	flagCompactionWorkers      = "compaction-workers"
+	flagEventsThreshold        = "etcd-events-threshold"
+	flagActiveDeadlineDuration = "active-deadline-duration"
+
+	// Secrets controller flags
+	flagSecretWorkers = "secret-workers"
+
+	// Etcd copy backups task controller flags
+	flagEtcdCopyBackupsTaskWorkers = "etcd-copy-backups-task-workers"
+)
+
+func AddControllerManagerFlags(c *config.ControllerManagerConfig) {
+	flag.StringVar(&c.MetricsAddr, flagMetricsAddr, c.MetricsAddr,
+		"The address the metric endpoint binds to.")
+	flag.BoolVar(&c.EnableLeaderElection, flagEnableLeaderElection, c.EnableLeaderElection,
+		"Enable leader election for controller manager. Enabling this will ensure there is only one active controller manager.")
+	flag.StringVar(&c.LeaderElectionID, flagLeaderElectionID, c.LeaderElectionID,
+		"Name of the resource that leader election will use for holding the leader lock")
+	flag.StringVar(&c.LeaderElectionResourceLock, flagLeaderElectionResourceLock, c.LeaderElectionResourceLock,
+		"Specifies which resource type to use for leader election. Supported options are 'endpoints', 'configmaps', 'leases', 'endpointsleases' and 'configmapsleases'.")
+	flag.BoolVar(&c.IgnoreOperationAnnotation, flagIgnoreOperationAnnotation, c.IgnoreOperationAnnotation,
+		"Specifies whether to ignore or honour the operation annotation on resources to be reconciled.")
+	flag.BoolVar(&c.DisableEtcdServiceAccountAutomount, flagDisableEtcdServiceAccountAutomount, c.DisableEtcdServiceAccountAutomount,
+		"If true then .automountServiceAccountToken will be set to false for the ServiceAccount created for etcd statefulsets.")
+	flag.BoolVar(&c.DisableLeaseCache, flagDisableLeaseCache, c.DisableLeaseCache,
+		"Disable cache for lease.coordination.k8s.io resources.")
+}
+
+func AddEtcdControllerFlags(c *config.EtcdControllerConfig) {
+	flag.IntVar(&c.Workers, flagEtcdWorkers, c.Workers,
+		"Number of worker threads for the etcd controller.")
+}
+
+func AddCustodianControllerFlags(c *config.CustodianControllerConfig) {
+	flag.IntVar(&c.Workers, flagCustodianWorkers, c.Workers,
+		"Number of worker threads for the custodian controller.")
+	flag.DurationVar(&c.SyncPeriod, flagCustodianSyncPeriod, c.SyncPeriod,
+		"Sync period of the custodian controller.")
+	flag.DurationVar(&c.EtcdMember.EtcdMemberNotReadyThreshold, flagEtcdMemberNotReadyThreshold, c.EtcdMember.EtcdMemberNotReadyThreshold,
+		"Threshold after which an etcd member is considered not ready if the status was unknown before.")
+	flag.DurationVar(&c.EtcdMember.EtcdMemberUnknownThreshold, flagEtcdMemberUnknownThreshold, c.EtcdMember.EtcdMemberUnknownThreshold,
+		"Threshold after which an etcd member is considered unknown.")
+}
+
+func AddCompactionLeaseController(c *config.CompactionLeaseControllerConfig) {
```

Comment:
There is no need to expose these methods if they are only going to be used from within `AddAllFlags`
Same comment applies to all other similar methods.